packages/atomic/src/components/search/result-lists/atomic-result-list/atomic-result-list.tsx

```diff
@@ -103,52 +86,38 @@ export class AtomicResultList
   /**
    * The expected size of the image displayed in the results.
    */
-  @Prop({reflect: true}) imageSize: ResultDisplayImageSize = 'icon';
+  @Prop({reflect: true, mutable: true}) imageSize: ResultDisplayImageSize =
+    'icon';
   /**
    * @deprecated use `imageSize` instead.
    */
   @Prop({reflect: true}) image: ResultDisplayImageSize = 'icon';
 
-  // to remove when `image` prop is removed;
-  private propToUseForImageSize: 'image' | 'imageSize' = 'imageSize';
-
   /**
    * Sets a rendering function to bypass the standard HTML template mechanism for rendering results.
    * You can use this function while working with web frameworks that don't use plain HTML syntax, e.g., React, Angular or Vue.
    *
    * Do not use this method if you integrate Atomic in a plain HTML deployment.
    *
-   * @param render
+   * @param resultRenderingFunction
    */
-  @Method() public async setRenderFunction(render: ResultRenderingFunction) {
-    this.renderingFunction = render;
-    this.assignRenderingFunctionIfPossible();
+  @Method() public async setRenderFunction(
+    resultRenderingFunction: ResultRenderingFunction
+  ) {
+    this.resultRenderingFunction = resultRenderingFunction;
   }
 
-  /**
-   * @internal
-   */
-  @Method() public async focusOnNextNewResult() {
-    this.resultListCommon.focusOnNextNewResult(this.resultListState);
+  public connectedCallback() {
+    this.settleImageSize();
   }
 
-  connectedCallback() {
-    // to remove when `image` prop is removed;
-    if (this.host.hasAttribute('image')) {
-      this.propToUseForImageSize = 'image';
-    }
+  // TODO: remove when `image` prop is removed;
+  private settleImageSize() {
     if (this.host.hasAttribute('image-size')) {
-      this.propToUseForImageSize = 'imageSize';
+      return;
     }
-  }
-
-  @Listen('scroll', {target: 'window'})
-  handleInfiniteScroll() {
-    if (
-      this.enableInfiniteScroll &&
-      this.resultListCommon.scrollHasReachedEndOfList
-    ) {
-      this.resultList.fetchMoreResults();
+    if (this.host.hasAttribute('image')) {
+      this.imageSize = this.image;
     }
   }
```

Comment:
Why are we doing it like that as opposed to having `imageSize` default to `undefined` and use `this.imageSize ?? this.image`?